### PR TITLE
refactor!: Remove grpc-web/rosetta and improve server performance

### PR DESCRIPTION
**Pull request by tkxkd0159**
## Description

closes: #1416

**After update:**
* Node operator need to remove rosetta/grpc-web config in `app.toml`
* Node operator need to run `envoy` proxy if they want to support grpc-web
* Node operator need to run `cosmos/rosetta` if they want to support Coinbase Rosetta API. This command also can be added in Finschia binary itself (`root.go`)

At this time, I also want to refactor our server code structure based on cosmos/cosmos-sdk#15041, cosmos/cosmos-sdk#16152.

## Motivation and context

1. We separate rosetta/grpc-web from main core to apply MSA. It will replace legacy code with latest one and improve the complication level of dependencies.
2. To improve query performance, we modify grpc-gateway to be concurrent.

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Review by zemyblue — APPROVED** [review]
Please introduce the reason why we remove this in this PR's description.

File: server/start.go [hunk 1]
```diff
@@ -338,98 +332,76 @@ func startInProcess(ctx *Context, clientCtx client.Context, appCreator types.App
...
-		}
-		errCh := make(chan error)
-
-		go func() {
-			if err := apiSrv.Start(config); err != nil {
-				errCh <- err
+		if srvcfg.GRPC.Enable {
+			_, port, err := net.SplitHostPort(srvcfg.GRPC.Address)
+			if err != nil {
+				return err
```
Comment (jaeseung-bae) [inline server/start.go:356]: Is it okay to nest `if srvcfg.GRPC.Enable` block in `if srvcfg.API.Enable`?
`if srvcfg.GRPC.Enable` was previously independent if clause.

Comment (tkxkd0159) [reply server/start.go]: I will revert that change!

**Review by jaeseung-bae — DISMISSED** [review]

File: testutil/network/util.go [hunk 2]
```diff
@@ -78,6 +78,15 @@ func startInProcess(cfg Config, val *Validator) error {
 		}
 	}
 
+	if val.AppConfig.GRPC.Enable {
```
Comment (0Tech) [inline testutil/network/util.go:81]: It would be nice to mention the relevant PR, cosmos/cosmos-sdk#14652, in the PR description.

**Review by zemyblue — APPROVED** [review]

File: server/start.go [hunk 3]
```diff
@@ -476,6 +391,94 @@ func genPvFileOnlyWhenKmsAddressEmpty(cfg *config.Config) *pvm.FilePV {
 	return nil
 }
 
+func startGrpcServer(
```
Comment (0Tech) [inline server/start.go:394]: cosmos/cosmos-sdk#16152

**Review by 0Tech — APPROVED** [review]

File: server/start.go (hunk 3 above)
Comment (tkxkd0159) [reply server/start.go:394]: I tried to minimize changes. But it's probably better to apply full refactoring this time. I will apply this PR as well.

Comment (0Tech) [reply server/start.go:394]: You may state your intention, and create an issue for the later refactoring. However, no one will stop you if you want to apply it right now.

**Review by 0Tech — APPROVED** [review]

**Pull request merged**